src/test/java/org/xwiki/git/script/GitScriptServiceTest.java

```diff
@@ -55,6 +56,54 @@
 
     private File testRepository;
 
+    @Unstable
```

Comment:
Same error as Usama... I don't understand why you both adding this annotation. Could you explain?

Where did you find the definition of it and the doc about it that would suggest to add it here? I'd like to see if there's something to improve.

Thx

Reply:
Hm, I thought we add this with any new method introduced (test methods as well). But I see now we dont need it for tests.

Reply:
Yes but why did you think we would need unstable? What is the purpose of it? you need to understand everything you do! And if something seems weird you need to question it. That’s a good way to progress.

Reply:
We use `@Unstable` notation to show in the backward-compatibility reports that this is a new API which will change quite frequently (as I read on Dev Practices). We use it for new methods or APIs. Now why I shouldn't have used it here is because it's not a Public API (which I forgot to remember from the Dev Practices) and also because tests are not the new API itself but to just test them. There's a lot of stuff to remember and I try as much as I can.

Reply:
Yes there's a lot! Indeed a test is not a public user API. We use @Unstable for public APIs only (not just for APIs). This allow us to stabilize the API and not break backward compatibility since user must use unstable public API at their own risks.

There's a lot but this practice of handling backward compatibility is really important and novel (you won't learn it elsewhere). So if you fully understand how we do this, you'll have learnt a lot more than to learn how to code in java or write unit tests since all that you'll learn anyway and it' pretty basic and obvious :) I highly recommend that you spend some time understand that including how we use aspects for it.

Reply:
Yep @vmassol this is actually giving me immensely useful knowledge which I never had in the past (by programming for a few years in different languages). I am not being very grateful I know, but it has been a steep learning curve for me. I admit that my reading and memory ability is not very good but I will try to gain your trust :)
And just to let you know, I've watched your speech on youtube about revapi, mutation testing (and more), but I only read the slides since I don't understand french. Also I saw your slides on slideshare about coverage testing. So my point is, I am not just trying to find something to copy paste as most students do. I understand what I can and always I will be glad to correct myself. I don't mean I unlike your criticism, I do appreciate your criticism as it motivates me to find new things.

Reply:
Thanks for your message @Haxsen. Just want to say that while I can be tough, there's really nothing personal and I really appreciate you sticking with me. TBH I'm sure I'm not doing the best mentoring job I could. I'm a bit short on time and I'm sure that I don't spend enough time explaining things, and there's a lot of things. They're all written down on dev.xwiki.org but there's really a lot. It probably takes about 6 months to understand them all so it's understandable that you cannot grasp everything at once.

I appreciate you checking out the videos. Some more info on this:
* You can check all talks and presentations I gave on my blog  (https://massol.net), at https://massol.myxwiki.org/xwiki/bin/view/Activities/Data/
* There's also a list of talks at https://dev.xwiki.org/xwiki/bin/view/PresentationIdea/. For ex related to revapi and mutation, in English, the last one is at https://massol.myxwiki.org/xwiki/bin/view/Activities/Data/BruJUG2020/. There was no recording though. Now for a recording, there's the FOSDEM one at https://massol.myxwiki.org/xwiki/bin/view/Activities/Data/FOSDEM2018JavaTestingNext/

Thanks!

Reply:
That's great, thanks :)

Reply:
BTW All of those lectures are extremely helpful, specially the BruJUG2020. I never had such knowledge.
Thanks a lot and keep them going :)